### PR TITLE
TST: 32bit dtype compat #36579

### PR DESCRIPTION
**Pull request by fangchenli**
Part of #36579

### CODE REVIEW

File: pandas/tests/indexes/test_base.py [hunk 1]
```diff
@@ -2607,7 +2608,8 @@ def construct(dtype):
         ex1 = np.array([0, 3, 1, 4, 2, 5] * 2, dtype=np.intp)
         ex2 = np.array([], dtype=np.intp)
         tm.assert_numpy_array_equal(result[0], ex1)
-        tm.assert_numpy_array_equal(result[1], ex2.astype(np.int64))
+        expected_dtype = np.int64 if IS64 else np.int32
+        tm.assert_numpy_array_equal(result[1], ex2.astype(expected_dtype))
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py:2612]: same

Comment (jreback) [reply pandas/tests/indexes/test_base.py:2612]: actually if you just remove the astype this should be correct

File: pandas/tests/test_algos.py [hunk 2]
```diff
@@ -1545,7 +1545,10 @@ def test_lookup_nan(self, writable):
         xs.setflags(write=writable)
         m = ht.Float64HashTable()
         m.map_locations(xs)
-        tm.assert_numpy_array_equal(m.lookup(xs), np.arange(len(xs), dtype=np.int64))
+        expected_dtype = np.int64 if IS64 else np.int32
```
Comment (jreback) [inline pandas/tests/test_algos.py:1548]: same

File: pandas/tests/test_algos.py [hunk 3]
```diff
@@ -1578,7 +1581,10 @@ def test_lookup_overflow(self, writable):
         xs.setflags(write=writable)
         m = ht.UInt64HashTable()
         m.map_locations(xs)
-        tm.assert_numpy_array_equal(m.lookup(xs), np.arange(len(xs), dtype=np.int64))
+        expected_dtype = np.int64 if IS64 else np.int32
```
Comment (jreback) [inline pandas/tests/test_algos.py:1584]: same

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

do we have a 32-bit build on CI? e.g. where were these failing

**Comment by fangchenli** [comment]
> thanks!
> 
> do we have a 32-bit build on CI? e.g. where were these failing

The 32-bit build is still work in progress https://github.com/pandas-dev/pandas/pull/35898. It's working but not integrated with our existing setup yet. I also saw those failures on Raspberry Pi.

**Pull request merged**